### PR TITLE
Split 'System/Utils' into header and source

### PR DESCRIPTION
**Pull request by vittorioromeo**
No real need for these functions to be `inline`. We can isolate a few includes in the `.cpp` file.

### CODE REVIEW

File: src/SFML/System/Utils.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+#include <string>
+
+
+namespace sf
+{
+std::string toLower(std::string str)
+{
+    for (char& c : str)
+        c = static_cast<char>(std::tolower(static_cast<unsigned char>(c)));
+
```
Comment (ChrisThrasher) [inline src/SFML/System/Utils.cpp:42]:
```suggestion
```

File: src/SFML/System/Utils.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    return str;
+}
+
+std::string formatDebugPathInfo(const std::filesystem::path& path)
+{
+    std::ostringstream oss;
+
```
Comment (ChrisThrasher) [inline src/SFML/System/Utils.cpp:49]:
```suggestion
```

File: src/SFML/System/Utils.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    oss << "    Provided path: " << path << '\n' //
+        << "    Absolute path: " << std::filesystem::absolute(path);
+
```
Comment (ChrisThrasher) [inline src/SFML/System/Utils.cpp:52]:
```suggestion
```

**Review by ChrisThrasher — APPROVED** [review]
Funny you bring this up. [My second SFML PR](https://github.com/SFML/SFML/pull/1907) was adding `toLower` to reduce some duplicate code. At the time I didn't implement it in a .cpp file purely out of a laziness haha. I somehow got away with putting it all in a header file

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Comment by vittorioromeo** [comment]
SFMLBuildMaster: Build this please.

**Comment by ChrisThrasher** [comment]
Rebased on `master` and fixed a minor conflict. This is ready to merge.

**Comment by ChrisThrasher** [comment]
SFMLBuildMaster: Build this please.

**Review by kimci86 — APPROVED** [review]

**Comment by eXpl0it3r** [comment]
I'm not that certain about this change.

We're now exporting symbols, which could to some extend be considered part of the public API, even if it's not exposed the same way. The inline guaranteed that the function was copied into the specific module before and thus didn't need to be exported...

**Pull request merged**